sdk/core/azure-core-amqp/src/models/amqp_message.cpp

```diff
@@ -455,24 +455,24 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
         // described body.
         AmqpDescribed describedBody(
             static_cast<std::uint64_t>(AmqpDescriptors::DataAmqpValue), message.m_amqpValueBody);
-        auto serializedBodyValue = AmqpValue::Serialize(describedBody);
+        auto serializedBodyValue = AmqpValue::Serialize(static_cast<AmqpValue>(describedBody));
```

Comment:
Why do we need these static casts? Is it possible to design the APIs to avoid the need for these as they tend to be error-prone. Should `Serialize` be able to handle different types that can be treated as `AmqpValue`?

Reply:
To work around what I believe is a GCC9 bug, I had to make the `AmqpXxx` to `AmqpValue` conversion explicit (as best as I can figure, the compiler was chosing an implicit conversion to `AmqpValue` over a copy constructor), which means that I needed to use static cast to force the conversion. `AmqpValue::Serialize` serializes an `AmqpValue`, not an `AmqpXxx`. 

I'd rather not add specializations to `AmqpValue::Serialize` for each of the possible `AmqpXxx` types because (a) I think it confuses the code, and (b) because the conversion between `AmqpXxx` and `AmqpValue` is actually a fairly significant transformation - it's not a simple cast.